token-metadata/program/src/state.rs

```diff
@@ -176,13 +219,34 @@ pub struct UseAuthorityRecord {
     pub bump: u8,
 }
 
+impl Default for UseAuthorityRecord {
+    fn default() -> Self {
+        UseAuthorityRecord {
+            key: Key::UseAuthorityRecord,
+            allowed_uses: 0,
+            bump: 255,
+        }
+    }
+}
+
+impl TokenMetadataAccount for UseAuthorityRecord {
+    fn key() -> Key {
+        Key::UseAuthorityRecord
+    }
+
+    fn size() -> usize {
+        USE_AUTHORITY_RECORD_SIZE
+    }
+}
+
 impl UseAuthorityRecord {
     pub fn from_account_info(a: &AccountInfo) -> Result<UseAuthorityRecord, ProgramError> {
```

Comment:
Why can't `from_account_info()` also be a deefault implementation in the  `TokenMetadataAccount` trait?  It seems like it always calls `safe_deserialize()` then check owner.  Maybe I'm missing some other detail though.

Reply:
It can but it makes things a bit more messy as it requires manually specifying the account type in a lot of places and it also causes issues with [this](https://github.com/metaplex-foundation/metaplex-program-library/blob/ac94ca76c9f1e371cee2e8e169435c209b48f150/token-metadata/program/src/state.rs#L285) because of the other trait derived on these accounts, `MasterEdition`.

Reply:
I figure out how to get around the issues with the function and added in all the types, so this works.

Reply:
Ok great!